scripts/testnet/include/config.sh

```diff
@@ -131,10 +131,49 @@ updateNodeConfig() {
      sed -i '/\[Antiflood\]/,/\[Logger\]/ s/true/false/' config_observer.toml
   fi
 
+  updateConfigsForStakingV4
+
   echo "Updated configuration for Nodes."
   popd
 }
 
+updateConfigsForStakingV4() {
+  config=$(cat enableEpochs.toml)
+
+  echo "Updating staking v4 configs"
+
+  # Get the StakingV4Step3EnableEpoch value
+  staking_enable_epoch=$(echo "$config" | awk -F '=' '/  StakingV4Step3EnableEpoch/{gsub(/^[ \t]+|[ \t]+$/,"", $2); print $2; exit}')
+  # Count the number of entries in MaxNodesChangeEnableEpoch
+  entry_count=$(echo "$config" | awk '/MaxNodesChangeEnableEpoch/,/\]/{if ($0 ~ /\{/) {count++}} END {print count}')
+
+  # Check if entry_count is less than 2
+  if [ "$entry_count" -lt 2 ]; then
+    echo "Not enough entries found to update"
+  else
+    # Find all entries in MaxNodesChangeEnableEpoch
+    all_entries=$(awk '/MaxNodesChangeEnableEpoch/,/\]/{if ($0 ~ /^[[:space:]]*\{/) {p=1}; if (p) print; if ($0 ~ /\]/) p=0}' enableEpochs.toml | grep -vE '^\s*#' | sed '/^\s*$/d')
+
+    # Get the index of the entry with EpochEnable equal to StakingV4Step3EnableEpoch
+    index=$(echo "$all_entries" | grep -n "EpochEnable = $staking_enable_epoch" | cut -d: -f1)
```

Comment:
what happens if there staking enable epoch is not found in an entry?

Reply:
Right, very good catch, fixed it.